pkg/funky/server.go

```diff
@@ -0,0 +1,174 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package funky
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// Server an interface for managing function servers
+type Server interface {
+	IsIdle() bool
+	SetIdle(bool)
+	GetPort() uint16
+	Invoke(input Message) (io.ReadCloser, error)
```

Comment:
Both input and return types should be `*Message` reflecting the fact that servers accept and return the same message structure consisting of context and payload. 

Making it a pointer saves a few bytes of memory and removes the illusion that those messages are immutable.

Reply:
This should also make `router.Delegate()` implementation much shorter and simpler.